lldb/include/lldb/Target/Process.h

```diff
@@ -614,6 +614,8 @@ class Process : public std::enable_shared_from_this<Process>,
     return error;
   }
 
+  virtual void DidLoadCore() {}
```

Comment:
We could avoid adding the DidLoadCore and re-use the existing Process DidAttach:
```
class Process {
...
  virtual void DidAttach(ArchSpec &process_arch) { process_arch.Clear(); }
```

Reply:
Good point! At least for core file process plugins, the `DidAttach` override shouldn't be implemented so we should be good.

Reply:
Yeah, I checked and none of the core file plug-ins currently override the DidAttach() (mach-o, ELF or minidump).

Reply:
Would it be worth to rename that function to `DidAttachOrLoadCore` (or just `DidAttachOrLoad`) to make it obvious that it handles both cases? I'm all for reusing `DidAttach` but it does seem a bit weird for a core file process.

Reply:
I think it makes sense in the process plug-in landscape. We are attaching to a core file image from a process plug-in perspective. I would rather add a `DidLoadCore()` over renaming to `DidAttachOrLoadCore()` if we do need to make a change though.

Reply:
I agree with @clayborg: attaching to a process and loading a core file are mutually exclusive actions so I can use re-using `DidAttach` in core file process plugins. However, if we rename `DidAttach` to `DidAttachOrLoadCore` then that would be called for non-corefile process plugins and I think that would be confusing to that a "LoadCore" mention in the context of the regular process.

@JDevlieghere if you think that using `DidAttach` in the core file process plugin context is confusing, I'm fine going back to my original implementation using `DidLoadCore`.

Reply:
Okay, so if semantically this doesn't make sense (which is what I'm hearing) then I don't why we would reuse `DidAttach` (besides it just happens to be there and unused in the core case) and we should have a `DidLoadCore` as per the current patch.